### ISSUE TITLE
Replace jcryption with SSL/TLS

### ISSUE THREAD

**Issue by slimm609** [nilsteampassnet/TeamPass#373]
jcryption over HTTP is not a replacement for SSL especially for a password manager.  A password manager is one of the systems that needs to be the most secure since it houses the passwords for other systems and accounts.   

jcryption SHOULD be removed and replaced by SSL/TLS enforcement. 

jcryption over HTTP is a giant man-in-the-middle vulnerability waiting to happen.  A user CAN NOT trust the javascript that is sent to the user because it is sent over HTTP.   An attacker can inject his own javascript into the stream to the end user would never know.   There is also no way to prevent this without SSL...

--excerpt from link below--
SSL/TLS is so much more than just encryption. In fact, SSL doesn't even have to be encrypted at all (the ...WITH_NULL cipher suites). But here are some of the features besides encryption:

server authentication
client authentication
resumable sessions
negotiable cipher suites
protection against man-in-the-middle attacks
perfect forward secrecy (if you use Diffie-Hellman or Elliptic Curve Diffie-Hellman key exchange)
...
SSL/TLS on a high-level view can be thought of as a mixture of asymmetric cryptography (to establish a session key) and symmetric cryptography (the "encryption" part that uses the formerly established session key). The big advantage over let's say just using AES is that you don't need to establish the keys initially. In your case, how would you make sure that client and server are using the same key? How would you establish it? How would you secure it?

TLS takes care of all this for you. See also this post for more thoughts on a similar topic. 

http://stackoverflow.com/questions/6932086/javascript-aes-encryption-vs-ssl

http://stackoverflow.com/questions/6658557/alternative-to-ssl-manual-encryption/6658596#6658596

http://stackoverflow.com/questions/1750214/can-i-use-js-encryption-instead-of-ssl-for-credit-card-payments

in short... Please, please, please remove jcryption entirely and enforce SSL.  I submitted a patch a while back for enforcing SSL but it didn't make it in.
